tests/geo/test_magneticfields.py

```diff
@@ -0,0 +1,115 @@
+import numpy as np
+import unittest
+
+from simsopt.geo.magneticfieldclasses import ToroidalField, ScalarPotentialRZMagneticField, CircularCoilXY
+from simsopt.geo.curvexyzfourier import CurveXYZFourier
+from simsopt.geo.magneticfield import MagneticFieldSum
+from simsopt.geo.curverzfourier import CurveRZFourier
+from simsopt.geo.curvehelical import CurveHelical
+from simsopt.geo.biotsavart import BiotSavart
+
+class Testing(unittest.TestCase):
+
+    def test_toroidal_field(self):
+        R0test    = 1.3
+        B0test    = 0.8
+        pointVar  = 1e-2
+        npoints   = 20
+        # point locations
+        points    = np.asarray(npoints * [[-1.41513202e-03,  8.99999382e-01, -3.14473221e-04]])
+        points   += pointVar * (np.random.rand(*points.shape)-0.5)
+        # Bfield from class
+        Bfield    = ToroidalField(R0test,B0test)
+        Bfield.set_points(points)
+        B1        = Bfield.B()
+        dB1_by_dX = Bfield.dB_by_dX()
+        # Bfield analytical
+        B2        = np.array([(B0test*R0test/(point[0]**2+point[1]**2))*np.array([-point[1], point[0], 0.]) for point in points])
+        dB2_by_dX = np.array([(B0test*R0test/((point[0]**2+point[1]**2)**2))*np.array([[2*point[0]*point[1], point[1]**2-point[0]**2, 0],[point[1]**2-point[0]**2, -2*point[0]*point[1], 0],[0,0,0]]) for point in points])
+        # Verify
+        assert np.allclose(B1, B2)
+        assert np.allclose(dB1_by_dX, dB2_by_dX)
```

Comment:
Could add a check for symmetry of gradient and also that divergence is zero

Reply:
Check added for the divergence of the magnetic field. Could you be specific about which symmetry you were looking for?

Reply:
For the fields that we get from the Biot Savart law for coils, we know that the 3x3 matrix `A = \nabla B` is symmetric. Is this the case here as well?

Reply:
I am not aware of such a symmetry.

P.S: Happy Easter!

Reply:
This symmetry should indeed be present here. This is a vacuum field so B = grad phi, hence grad B = grad grad phi which is manifestly symmetric.

Reply:
Right! Test added